packages/contracts-bedrock/test/invariants/symbolic/OptimismSuperchainERC20.t.sol

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { Test } from "forge-std/Test.sol";
+import "forge-std/Test.sol";
+
+import "src/L2/OptimismSuperchainERC20.sol";
+import { OptimismSuperchainERC20 } from "src/L2/OptimismSuperchainERC20.sol";
+import { SymTest } from "halmos-cheatcodes/src/SymTest.sol";
+import { L2ToL2CrossDomainMessenger } from "src/L2/L2ToL2CrossDomainMessenger.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { ERC1967Proxy } from "@openzeppelin/contracts-v5/proxy/ERC1967/ERC1967Proxy.sol";
+import { MockL2ToL2Messenger } from "./MockL2ToL2Messenger.sol";
+import "src/L2/L2ToL2CrossDomainMessenger.sol";
+
+// TODO: Move to a advanced tests helper folder
+interface IHevm {
+    function chaind(uint256) external;
+
+    function etch(address addr, bytes calldata code) external;
+
+    function prank(address addr) external;
+
+    function deal(address, uint256) external;
+
+    function deal(address, address, uint256) external;
+}
+
+contract HalmosTest is SymTest, Test { }
+
+contract OptimismSuperchainERC20_SymTest is HalmosTest {
+    uint256 internal constant CURRENT_CHAIN_ID = 1;
+    uint256 internal constant ZERO_AMOUNT = 0;
+    MockL2ToL2Messenger internal constant MESSENGER = MockL2ToL2Messenger(Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER);
+    IHevm hevm = IHevm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+
+    address internal remoteToken = address(bytes20(keccak256("remoteToken")));
+    string internal name = "SuperchainERC20";
+    string internal symbol = "SUPER";
+    uint8 internal decimals = 18;
+    address internal user = address(bytes20(keccak256("user")));
+    address internal target = address(bytes20(keccak256("target")));
+
+    OptimismSuperchainERC20 public superchainERC20Impl;
+    OptimismSuperchainERC20 internal optimismSuperchainERC20;
+
+    constructor() {
+        superchainERC20Impl = new OptimismSuperchainERC20();
+        optimismSuperchainERC20 = OptimismSuperchainERC20(
+            address(
+                new ERC1967Proxy(
+                    address(superchainERC20Impl),
+                    abi.encodeCall(OptimismSuperchainERC20.initialize, (remoteToken, name, symbol, decimals))
+                )
+            )
+        );
+
+        // Etch the mocked L2 to L2 Messenger because the `TSTORE` opcode is not supported, and also due to issues with
+        // `encodeVersionedNonce()`
```

Comment:
we mock the whole crosschain messaging, so it's not only for these issue, it's the whole test design

Reply:
True, will add on the comment